src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -4714,6 +4714,22 @@ std::set<String> MergeTreeData::getPartitionIdsAffectedByCommands(
     return affected_partition_ids;
 }
 
+std::unordered_set<String> MergeTreeData::getAllPartitionIds() const
+{
+    auto lock = lockParts();
+    std::unordered_set<String> res;
+    String prev_id;
```

Comment:
I'm not sure if parts can change, but I assume they can't because you have a lock.
Maybe use `std::string_view` here so you don't copy every time?